File: Web/src/utils/countries.ts
/** Country option for UI display. Only `code` is sent to the backend. */
export interface CountryOption {
  name: string;
  code: string;
}

/**
 * Static ISO 3166-1 alpha-2 country list.
 * The UI shows `name` (e.g. "Pakistan") but every API call uses `code` (e.g. "pk").
 */
export const COUNTRIES: CountryOption[] = [
  { name: "United States", code: "us" },
  { name: "United Kingdom", code: "gb" },
  { name: "Canada", code: "ca" },
  { name: "Australia", code: "au" },
  { name: "Germany", code: "de" },
  { name: "France", code: "fr" },
  { name: "Netherlands", code: "nl" },
  { name: "Sweden", code: "se" },
  { name: "Norway", code: "no" },
  { name: "Denmark", code: "dk" },
  { name: "Finland", code: "fi" },
  { name: "Ireland", code: "ie" },
  { name: "Switzerland", code: "ch" },
  { name: "Austria", code: "at" },
  { name: "Belgium", code: "be" },
  { name: "Spain", code: "es" },
  { name: "Italy", code: "it" },
  { name: "Portugal", code: "pt" },
  { name: "Poland", code: "pl" },
  { name: "Czech Republic", code: "cz" },
  { name: "India", code: "in" },
  { name: "Pakistan", code: "pk" },
  { name: "Bangladesh", code: "bd" },
  { name: "United Arab Emirates", code: "ae" },
  { name: "Saudi Arabia", code: "sa" },
  { name: "Singapore", code: "sg" },
  { name: "Malaysia", code: "my" },
  { name: "Indonesia", code: "id" },
  { name: "Philippines", code: "ph" },
  { name: "Japan", code: "jp" },
  { name: "South Korea", code: "kr" },
  { name: "China", code: "cn" },
  { name: "Hong Kong", code: "hk" },
  { name: "New Zealand", code: "nz" },
  { name: "South Africa", code: "za" },
  { name: "Nigeria", code: "ng" },
  { name: "Kenya", code: "ke" },
  { name: "Egypt", code: "eg" },
  { name: "Brazil", code: "br" },
  { name: "Mexico", code: "mx" },
  { name: "Argentina", code: "ar" },
  { name: "Colombia", code: "co" },
  { name: "Chile", code: "cl" },
  { name: "Turkey", code: "tr" },
  { name: "Israel", code: "il" },
  { name: "Qatar", code: "qa" },
  { name: "Kuwait", code: "kw" },
  { name: "Romania", code: "ro" },
  { name: "Hungary", code: "hu" },
  { name: "Greece", code: "gr" },
  { name: "Ukraine", code: "ua" },
].sort((a, b) => a.name.localeCompare(b.name));

export function getCountryName(code: string): string {
  const normalized = code.toLowerCase();
  return COUNTRIES.find((country) => country.code === normalized)?.name ?? code.toUpperCase();
}

export function getCountryCode(nameOrCode: string): string {
  const normalized = nameOrCode.toLowerCase();
  const byCode = COUNTRIES.find((country) => country.code === normalized);
  if (byCode) {
    return byCode.code;
  }
  const byName = COUNTRIES.find((country) => country.name.toLowerCase() === normalized);
  return byName?.code ?? normalized;
}

File: Mobile/lib/core/data/countries.dart
class CountryOption {
  const CountryOption({required this.name, required this.code});

  final String name;
  final String code;
}

/// Static ISO country list. UI shows names, API receives codes only.
final List<CountryOption> countries = [
  const CountryOption(name: 'United States', code: 'us'),
  const CountryOption(name: 'United Kingdom', code: 'gb'),
  const CountryOption(name: 'Canada', code: 'ca'),
  const CountryOption(name: 'Australia', code: 'au'),
  const CountryOption(name: 'Germany', code: 'de'),
  const CountryOption(name: 'France', code: 'fr'),
  const CountryOption(name: 'Netherlands', code: 'nl'),
  const CountryOption(name: 'Sweden', code: 'se'),
  const CountryOption(name: 'Norway', code: 'no'),
  const CountryOption(name: 'Denmark', code: 'dk'),
  const CountryOption(name: 'Finland', code: 'fi'),
  const CountryOption(name: 'Ireland', code: 'ie'),
  const CountryOption(name: 'Switzerland', code: 'ch'),
  const CountryOption(name: 'Austria', code: 'at'),
  const CountryOption(name: 'Belgium', code: 'be'),
  const CountryOption(name: 'Spain', code: 'es'),
  const CountryOption(name: 'Italy', code: 'it'),
  const CountryOption(name: 'Portugal', code: 'pt'),
  const CountryOption(name: 'Poland', code: 'pl'),
  const CountryOption(name: 'Czech Republic', code: 'cz'),
  const CountryOption(name: 'India', code: 'in'),
  const CountryOption(name: 'Pakistan', code: 'pk'),
  const CountryOption(name: 'Bangladesh', code: 'bd'),
  const CountryOption(name: 'United Arab Emirates', code: 'ae'),
  const CountryOption(name: 'Saudi Arabia', code: 'sa'),
  const CountryOption(name: 'Singapore', code: 'sg'),
  const CountryOption(name: 'Malaysia', code: 'my'),
  const CountryOption(name: 'Indonesia', code: 'id'),
  const CountryOption(name: 'Philippines', code: 'ph'),
  const CountryOption(name: 'Japan', code: 'jp'),
  const CountryOption(name: 'South Korea', code: 'kr'),
  const CountryOption(name: 'China', code: 'cn'),
  const CountryOption(name: 'Hong Kong', code: 'hk'),
  const CountryOption(name: 'New Zealand', code: 'nz'),
  const CountryOption(name: 'South Africa', code: 'za'),
  const CountryOption(name: 'Nigeria', code: 'ng'),
  const CountryOption(name: 'Kenya', code: 'ke'),
  const CountryOption(name: 'Egypt', code: 'eg'),
  const CountryOption(name: 'Brazil', code: 'br'),
  const CountryOption(name: 'Mexico', code: 'mx'),
  const CountryOption(name: 'Argentina', code: 'ar'),
  const CountryOption(name: 'Colombia', code: 'co'),
  const CountryOption(name: 'Chile', code: 'cl'),
  const CountryOption(name: 'Turkey', code: 'tr'),
  const CountryOption(name: 'Israel', code: 'il'),
  const CountryOption(name: 'Qatar', code: 'qa'),
  const CountryOption(name: 'Kuwait', code: 'kw'),
  const CountryOption(name: 'Romania', code: 'ro'),
  const CountryOption(name: 'Hungary', code: 'hu'),
  const CountryOption(name: 'Greece', code: 'gr'),
  const CountryOption(name: 'Ukraine', code: 'ua'),
]..sort((a, b) => a.name.compareTo(b.name));

String getCountryName(String code) {
  final normalized = code.toLowerCase();
  return countries.firstWhere((c) => c.code == normalized, orElse: () => CountryOption(name: code.toUpperCase(), code: normalized)).name;
}

String? countryCodeFromName(String? name) {
  if (name == null || name.isEmpty) {
    return null;
  }
  final normalized = name.toLowerCase();
  for (final country in countries) {
    if (country.code == normalized || country.name.toLowerCase() == normalized) {
      return country.code;
    }
  }
  return null;
}
